corehq/warehouse/tests/test_integration.py

```diff
@@ -0,0 +1,104 @@
+from datetime import datetime, timedelta
+
+from django.test import TestCase
+
+from corehq.apps.users.models import WebUser, CommCareUser
+from corehq.apps.domain.models import Domain
+from corehq.form_processor.tests.utils import create_form_for_test, FormProcessorTestUtils
+
+from corehq.warehouse.models import (
+    UserStagingTable,
+    DomainStagingTable,
+    UserDim,
+    DomainDim,
+    FormStagingTable,
+    FormFact,
+)
+
+
+class FormFactIntegrationTest(TestCase):
+    '''
+    Tests a full integration of loading the FormFact table from
+    staging and dimension tables.
+    '''
+    domain = 'form-fact-integration-test'
+
+    @classmethod
+    def setUpClass(cls):
+        cls.domain_records = [
+            Domain(name=cls.domain, hr_name='One', creating_user_id='abc', is_active=True),
+        ]
+
+        for domain in cls.domain_records:
+            domain.save()
+
+        cls.user_records = [
+            # TODO: Handle WebUsers who have multiple domains
```

Comment:
this is going to be a pain. since webusers have multiple domains, it means we most likely have to create these records differently when first fetching them from couch